### DISCUSSION TITLE
"Out of Memory" crashes

### DISCUSSION THREAD

**Discussion by KatsPurr** [zero01101/openOutpaint#273] [General]
Let me start off by THANKING the creators of this extension. OpenOutpaint is literally the main tool I use with AI and I couldn't live without it. Let me also say that I am in no way a programmer or skilled with tech stuff. So although I can follow instructions and can have a rough idea of what's going on, I don't really know and just follow the steps. So if anyone replies, just keep that in mind haha. 

Anyhow so I made the grave mistake of having _git pull_ in my Automatic1111 bat file. And apparently it automatically updated to version v1.6.1 yesterday, which in the process broke OpenOutpaint which now crashes Automatic1111. Knowing that it might take a while until OpenOutpaint is updated, I scrambled to try and figure out how to solve the situation. Obviously disabling OpenOutpaint works for just getting Automatic1111 to stop crashing and do all the other basic stuff. But I CAN'T LIVE WITHOUT OPENOUTPAINT! Literally!

So then I tried downgrading Automatic1111 to the previous version (v1.6.0 / 5ef669d) thinking that surely THAT will solve it! But that didn't help either! It still crashes Automatic with the "Out of memory" message. I also tried reinstalling the /venv/ folder. That didn't help either.

Downgrading Automatic1111 to the version that was working in harmony with OpenOutpaint up until yesterday "should" work, right? Is there something I'm overlooking? Something I need to do in addition to using the _git reset --hard 5ef669d_ command in CMD? All I want is to make my Automatic1111 compatible with OpenOutpaint again. * cries *

**Comment by zero01101** [comment]
sorry for not paying much attention in discussions :| much obliged for the flattery however :)

soooo uh this one kinda stymies me; openOutpaint is literally just a tab in webUI with an iframe hosting an HTML page with some javascript and CSS, there's no additional requirements or dependencies in the python side of things, so it's practically no different than having another tab open as far as browser memory usage should go, especially with a "blank canvas"... if you disable/uninstall the extension and just use the standalone version ([base instructions begin here](https://github.com/zero01101/openOutpaint/wiki/Manual#webui-launch-script--commandline_args)) does it work as expected?

**Reply by KatsPurr** [reply]
No worries at all! And thank you for the reply!

Ok so it shouldn't crash the Automatic1111 huh? Interesting. Yeah the same thing happened during the update earlier in the fall as well. And once OpenOutpaint had been updated, it resolved everything. So not sure what's going on, but if it would be helpful for you to see the problem "in action", I'd be more than happy to capture video of what happens and upload to YouTube if it would be helpful in any way. Just let me know. None of the other extensions (even outdated ones) do this, only OpenOutpaint. So it's the only I have to completely disable whenever there is an Automatic1111 update.

As for the standalone version... when I was desperately searching for info, I did see come across the instructions for setting that up, but as a non techy person, it all looked too overwhelming, so I didn't dare touch it. 

I might just have to take the plunge though. Depends on whether this update issue can be resolved any time soon. Right now I'm trying to just manage with the standard Automatic1111 inpainting tools, but they are SOOOOOOO bad and primitive compared to OpenOutpaint. You guys really made such an amazing tool! (sorry for all the gushing)

**Reply by KatsPurr** [reply]
So I have some news to report that might be useful.

Remember when I was telling you I use Chrome and you suggested trying Edge instead? And then I did, and I got OpenOutpaint to work again after that?

Well... Now I'm sad to say that Edge has stopped working as well and is behaving exactly the same way Chrome did: Crashing Gradio and giving the "out of memory error". 

So then I started to think to myself: What is something that would not immediately cause problems but might accumulate over time and cause problems: CACHE! But I tried clearing the cache and it didn't have any effect what so ever. 

So what helped last time? A new browser.

So just for the heck of it, I installed Firefox, fired up Automatic1111 and OpenOutpaint WORKED.

FOR NOW.

We left off with our conversation in the beginning of December, so I got about a month's worth of using OpenOutpaint on a "fresh browser" to be able to use it. Considering I use it at the same pace every day, I am going to predict that in about a month's time Firefox will stop working also.

It's as if there is something that gradually messes things up or clogs up the system over time. If it isn't cache, I don't know what it could be. But this might give some kind of indicator of what the root cause is.

One "warning symptom" that Edge had before it stopped working, was that as soon as I started up Automatic1111 it would take about 10 seconds of working normally, then it would go into "not responding" mode for a few seconds but recover from it and I was able to continue using it. The timing and pacing was exactly like a Gradio "out of memory" crash, but without actually crashing but instead managing to recover before going completely dead. Again, this ONLY happens with OpenOutpaint and not with any other extension.

So I'm not sure if these symptoms give you any new ideas of what could be causing this. And even though it works on Firefox for now, I'm sure that in about a month I'll be back with a video showing how it crashes with all 3 browsers in an identical way.
